### PR TITLE
Pbw 6756

### PR DESCRIPTION
**Pull request by gramcha**

### CODE REVIEW

File: js/components/controlBar.js [hunk 1]
```diff
@@ -259,8 +259,13 @@ var ControlBar = React.createClass({
         "oo-volume-bar": true,
         "oo-on": turnedOn
       });
-      var barStyle = { backgroundColor: this.props.skinConfig.controlBar.volumeControl.color ? this.props.skinConfig.controlBar.volumeControl.color : this.props.skinConfig.general.accentColor };
-
+      var barStyle = turnedOn === true ? ({
```
Comment (pilievOoyala) [inline js/components/controlBar.js]: Can you please change this to use if/else?  It's very difficult to follow and very likely we will mess something up if we have to edit it.

**Comment by pilievOoyala** [comment]
Please add unit tests

**Comment by pilievOoyala** [comment]
Also, just make sure to update the submodule when the other repo's PR is merged.

File: js/components/controlBar.js (hunk 1 above)
Comment (gramcha) [reply js/components/controlBar.js]: Modified as you said.

**Comment by gramcha** [comment]
@pilievOoyala 
Added unit tests. Please check it out.

**Comment by pilievOoyala** [comment]
Looks good. We need two approvals though. @aeng7 or @jpfranco can you also review?

File: js/components/controlBar.js [hunk 2]
```diff
@@ -259,6 +259,47 @@ var ControlBar = React.createClass({
...
+          barStyle = {backgroundColor: this.props.skinConfig.controlBar.volumeControl.inactiveColor ?
+                  this.props.skinConfig.controlBar.volumeControl.inactiveColor : this.props.skinConfig.general.inactiveAccentColor};
+      }
+      volumeBars.push(<a data-volume={(i + 1) / 10} className={volumeClass} key={i}
+        style={barStyle}
+        onClick={this.handleVolumeClick}
+        aria-hidden="true"></a>);
+    }
+
+    var volumeSlider = <div className="oo-volume-slider"><Slider value={parseFloat(this.props.controller.state.volumeState.volume)}
```
Comment (aeng7) [inline js/components/controlBar.js]: How does this new code interact with the old code to render the volume slider? I don't see any code removed, so was curious about the matter.

File: tests/components/controlBar-test.js [hunk 3]
```diff
@@ -234,6 +234,125 @@ describe('ControlBar', function () {
...
+              "inactiveColor": "red"}
+        ;
+
+      var mockProps = {
+          isLiveStream: false,
+          controller: mockController,
+          skinConfig: volumebarSkinConfig
+      };
+
+        var DOM = TestUtils.renderIntoDocument(
```
Comment (aeng7) [inline tests/components/controlBar-test.js]: Inconsistent spacing.

File: tests/components/controlBar-test.js [hunk 4]
```diff
@@ -234,6 +234,125 @@ describe('ControlBar', function () {
...
+          },
+          handleMuteClick: function() {muteClicked = false;},
+          setVolume: function(volume) {newVolume = volume;}
+      };
+
+      var volumebarSkinConfig = Utils.clone(skinConfig);
+      volumebarSkinConfig.controlBar.volumeControl =
+        {"color": "green",
+              "inactiveColor": "red"}
+        ;
```
Comment (aeng7) [inline tests/components/controlBar-test.js]: Inconsistent spacing.

File: tests/components/controlBar-test.js [hunk 5]
```diff
@@ -234,6 +234,125 @@ describe('ControlBar', function () {
...
+          },
+          handleMuteClick: function() {muteClicked = false;},
+          setVolume: function(volume) {newVolume = volume;}
+      };
+
+      var volumebarSkinConfig = Utils.clone(skinConfig);
+      volumebarSkinConfig.controlBar.volumeControl =
+          {"color": "green",
+              "inactiveColor": "red"}
+      ;
```
Comment (aeng7) [inline tests/components/controlBar-test.js]: Inconsistent spacing.

**Comment by jpfranco** [comment]
@gramcha unfortunately it seems that some of this will have to be reworked. We moved the volume controls to their own component (https://github.com/ooyala/html5-skin/pull/804), but this code is based off an older version. Will close this for now in order to prevent this from getting merged, but we should discuss what are the next steps for this fix with @aldigun

**Pull request closed without merging**